src/app/src/main/kotlin/org/imperial/mrc/hint/exceptions/HintException.kt

```diff
@@ -1,6 +1,10 @@
 package org.imperial.mrc.hint.exceptions
 
 import org.springframework.http.HttpStatus
+import java.net.URI
 
-open class HintException(val key: String, val httpStatus: HttpStatus = HttpStatus.INTERNAL_SERVER_ERROR) :
-        Exception("HintException with key $key")
+open class HintException(
+        val key: String,
+        val httpStatus: HttpStatus = HttpStatus.INTERNAL_SERVER_ERROR,
+        val adrUri: String? = null
```

Comment:
Could we avoid having error-specific information like `adrUrl` in the base exception class? If we need to have some additional info beyond key and httpStatus, could we handle those generically, eg. additionalArgs array? ...But anyway, it looks to me like the only place where adrUri is actually used, is in `HintExceptionHandler.handleAdrException` where the parameter is explicitly an `AdrException` anyway - so could we exclude it from the base class?